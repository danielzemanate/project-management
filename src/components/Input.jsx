import React from "react";

const Input = ({ label, name, defaultValue, type, required,disabled }) => {
  return (
    <label htmlFor={name} className="flex flex-col my-3 form-label">
      <span className="mb-2 text-xl font-semibold">{label}</span>
      <input
        data-testid="input-test"
        required={required}
        type={type}
        name={name}
        className="input px-3 form-control"
        defaultValue={defaultValue}
        disabled={disabled}
      />
      <div className="valid-feedback" data-testid="test-html-input">Correcto!</div>
      <div className="invalid-feedback">Introduzca un {name} valido.</div>
    </label>
  );
};

export default Input;
