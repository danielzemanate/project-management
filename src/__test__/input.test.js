import React from 'react';
import Input from 'components/Input';
import { render, screen, cleanup } from '@testing-library/react';


afterEach(cleanup);

it('renders okay', () => {
  render(<Input text='hola' loading={false} disabled={false} />);
  expect(screen.getByTestId('input-test')).toBeInTheDocument();
});

// it('shows text when not loading', () => {
//     render(<Input text='hola' loading={false} disabled={false} />);
//     expect(screen.getByTestId('input-test')).toHaveTextContent('hola');
//   });
  
  it('doesnt show text when loading', () => {
    render(<Input text='hola' loading={true} disabled={false} />);
    expect(screen.getByTestId('input-test')).not.toHaveTextContent('hola');
  });

  it('shows loading component when loading', () => {
    render(<Input text='hola' loading={true} disabled={false} />);
    expect(screen.getByTestId('test-html-input')).toBeInTheDocument();
  });

  it('is disabled when prop is passed', () => {
    render(<Input text='hola' loading={true} disabled={true} />);
    expect(screen.getByTestId('input-test')).toHaveAttribute('disabled');
  });

  it('is enabled when disabled prop is passed as false', () => {
    render(<Input text='hola' loading={true} disabled={false} />);
    expect(screen.getByTestId('input-test')).not.toHaveAttribute('disabled');
  });

  it('loads the svg html when loading is activated', () => {
    render(<Input text='hola' loading={true} disabled={false} />);
    expect(screen.getByTestId('input-test')).toMatchSnapshot();
  });