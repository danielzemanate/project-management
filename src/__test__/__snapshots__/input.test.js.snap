// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads the svg html when loading is activated 1`] = `
<input
  class="input px-3 form-control"
  data-testid="input-test"
  value=""
/>
`;
